setup/main.go

```diff
@@ -0,0 +1,1653 @@
+	// never modifies Postgres, the collector config, or anything else in the installed
+	// system
+	Check func(state *SetupState) (bool, error)
+	// Make changes to the system necessary for the check to pass, always prompting for
+	// user input before any change that modifies Postgres, the collector config, or
+	// anything else in the installed system
+	Run func(state *SetupState) error
+}
+
+const defaultConfigFile = "/etc/pganalyze-collector.conf"
+
+func main() {
+	steps := []*Step{
+		determinePlatform,
+		loadConfig,
+		saveAPIKey,
+		establishSuperuserConnection,
+		checkPostgresVersion,
+		checkReplicationStatus,
+		selectDatabases,
+		specifyMonitoringUser,
+		createMonitoringUser,
+		configureMonitoringUserPasswd,
+		applyMonitoringUserPasswd,
+		setUpMonitoringUser,
+		createPganalyzeSchema,
+		// TODO: this does not work right now--we check with has_schema_privilege, and after
+		// a revoke, the user still has the privilege
+		//revokePrivilegesFromMonitoringUser,
+		checkPgssAvailable,
+		createPgss,
+		enablePgss,
+
+		confirmLogInsightsSetup,
+		configureLogErrorVerbosity,
+		configureLogDuration,
+		configureLogStatement,
+		configureLogMinDurationStatement,
+		configureLogLinePrefix,
+		configureLogLocation,
+
+		confirmAutoExplainSetup,
+		checkUseLogBasedExplain,
+		createLogExplainHelper,
+		checkAutoExplainAvailable,
+		enableAutoExplain,
+
+		reloadCollector,
+		restartPg,
+		configureAutoExplain,
+		runPgSleep,
+	}
+
+	var setupState SetupState
+	var verbose bool
+	var logFile string
+	flag.StringVar(&setupState.ConfigFilename, "config", defaultConfigFile, "Specify alternative path for config file")
+	flag.BoolVar(&verbose, "verbose", false, "Include verbose logging output")
+	flag.StringVar(&logFile, "log", "", "Save output to log file (always includes verbose output)")
+	flag.Parse()
+	var logger Logger
+	if logFile != "" {
+		logger = NewLogger()
+		log, err := os.Create(logFile)
+		if err != nil {
+			fmt.Errorf("could not open %s for writes: %s", logFile, err)
+			return
+		}
+		defer log.Close()
+		outputBoth := io.MultiWriter(os.Stdout, log)
+		logger.StandardOutput = outputBoth
+		if verbose {
+			logger.VerboseOutput = outputBoth
+		} else {
+			logger.VerboseOutput = log
+		}
+	} else {
+		logger = NewLogger()
+		if verbose {
+			logger.VerboseOutput = os.Stdout
+		}
+	}
+	setupState.Logger = &logger
+
+	// TODO: check for root?
+
+	setupState.Log(`Welcome to the pganalyze collector installer!
+
+We will go through a series of steps to set up the collector to monitor your
+Postgres database. We will not make any changes to your database or system
+without confirmation.
+
+At a high level, we will:
+
+ 1. Configure a database user and helper functions for the collector, with minimal access
+ 2. Update the collector configuration file with these settings
+ 3. Set up the pg_stat_statements extension in your database for basic query performance monitoring
```

Comment:
I think we can drop the `basic` here (I know we use it in the UI, but I think its more clear if we don't have it)